group-anagrams/sunjae95.js

```diff
@@ -0,0 +1,23 @@
+/**
+ * @description
+ * brainstorming:
+ * brute force + hashtable
+ *
+ * time complexity: O(n* k log k)
+ * space complexity: O(n* k)
+ */
+var groupAnagrams = function (strs) {
+  const map = new Map();
+  const answer = [];
+
+  strs.forEach((str) => {
+    const convertedStr = str.split("").sort().join();
+    if (map.has(convertedStr))
+      map.set(convertedStr, map.get(convertedStr).concat(str));
+    else map.set(convertedStr, [str]);
+  });
+
+  map.forEach((value) => answer.push(value));
+
+  return answer;
```

Comment:
[안 중요] 요렇게 하면 더 깔끔하지 않을까 생각이 들었습니다. 11번째 줄도 필요가 없어지고요.

```suggestion
  return return [...map.values()];
```

Reply:
iterator는 spread operate가 가능하군요!